src/scanner.c

```diff
@@ -920,8 +934,12 @@ static UInt NextSymbol(void)
         c = GET_NEXT_CHAR();
     }
 
-  STATE(SymbolStartLine) = GetInputLineNumber();
-  STATE(SymbolStartPos) = GetInputLinePosition();
+    STATE(SymbolStartLine)[2] = STATE(SymbolStartLine)[1];
+    STATE(SymbolStartPos)[2] = STATE(SymbolStartPos)[1];
+    STATE(SymbolStartLine)[1] = STATE(SymbolStartLine)[0];
+    STATE(SymbolStartPos)[1] = STATE(SymbolStartPos)[0];
+    STATE(SymbolStartLine)[0] = GetInputLineNumber();
+    STATE(SymbolStartPos)[0] = GetInputLinePosition();
```

Comment:
Perhaps this should be moved into its own helper function, say, `static void RecordInputPosition(void)` or so?

Reply:
Yes, once I had to add 2 levels, I should have abstracted out.

Reply:
I also added some more comments, as it's a bit subtle what's going on around here.